lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -520,6 +521,10 @@ tokenize(String, Line, Column, Scope, Tokens) ->
           {error, Reason, String, Tokens};
         _ when Kind == alias ->
           tokenize_alias(Rest, Line, Column, Atom, Length, Ascii, Special, Scope, Tokens);
+        [$= | _] when HasBangOrQuestionMark, Kind == identifier ->
+          Msg = "use a space before = if the previous identifier ends with ! or ?",
```

Comment:
I would do something like:

```erlang
    io_lib:format("found an identifier ~ts, ending with ~ts, followed by =. It is unclear if you mean \"~ts ~ts=\" or \"~ts =". Please add a space before or after ~ts to remove the ambiguity", [Identifier, [Last], IdentifierWithoutLast, [Last], Identifier, [Last], [Last]])
```

That's because there are ways to solve the ambiguity, so we should be clear about them.